src/rgw/rgw_op.cc

```diff
@@ -1397,13 +1397,27 @@ void RGWListBuckets::execute()
 
   uint64_t max_buckets = s->cct->_conf->rgw_list_buckets_max_chunk;
 
+  rgw_user *accounts_user;
```

Comment:
why not use a reference and not a pointer?
why not initialize it here to s->user->user_id?

Reply:
Because this needs to wind up pointing to either "buckets_user" or "s->user->user_id".  references are designed to be immutable pointers, and here, this has to be mutable.